### PR TITLE
Add manipulation environments

### PR DESCRIPTION
**Pull request by SimpleGeometry**
This PR adds 4-5 manipulation environments of varying difficulties, many are similar in flavor. In particular, it adds a reach and grasp task, and then three pick and place tasks: two without bins ("pickplace_easy" and "pickplace_hard") and one with bins ("binpick"). Environment assets are adapted from Mujoco Menagerie's Franka Panda arm.

Environments work for MJX (positional is not advised/somewhat unstable still, and the other backends are much worse). Environments have been tested for collision/friction, stability, and speed, yielding around 10k-40k SPS on an A100 (depending on environment and network size). If helpful, I can provide htmls for visualizations.

Also, refactors utils.py slightly.

### CODE REVIEW

**Review by MichalBortkiewicz — CHANGES_REQUESTED** [review]
Minor problem with `update_goal` and perhaps we might abstract some part that is the same for all the tasks in a single file?

File: envs/arm_binpick.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+from jax import numpy as jnp
+
+
+# Observation space: 46 dims for the actual observation, 7 dims for goal position. See _get_obs() for details.
+# Action space:      9 dimensional, [-1, 1] each. The first 7 correspond to joint target angles (each joint has a different range, see
+#                    convert_action). The last 2 correspond to gripper finger positions.
+class ArmBinpick(PipelineEnv):
+    def __init__(self, backend="mjx", **kwargs):
+        # Load XML and manage simulation parameters
+        if backend == "mjx":
```
[the remark this answers is not in the document]
Comment (SimpleGeometry) [reply envs/arm_binpick.py:15]: Perhaps I should disable positional for now -- I'd say positional is half-working. I made a lot of changes to try stabilizing it with mixed success, so I'd like to hold on to the positional XMLs, but at the moment MJX is probably the only one that makes sense to use. Will edit.

File: envs/arm_binpick.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    # We deviate from the existing interface by allowing the goal to be modified directly,
+    # since this is useful for exploration during trajectory collection.
+    #
+    # Additionally, the goal is now virtual, and is no longer a physical object in simulation. While 
+    # a physical representation better for visualization of positional goals, many goals are not only
+    # positions. It could include orientation, velocity, timestep, gripper open-ness, etc. which cannot be 
+    # represented naturally with an object.
+    def update_goal(self, state: State, goal: jax.Array) -> State:
+        updated_info = {**info, "goal": goal} # Dictionary unpacking to return updated dict
+        return state.replace(info=updated_info)
```
[the remark this answers is not in the document]
Comment (SimpleGeometry) [reply envs/arm_binpick.py:115]: Oops, yeah, this function I hadn't tested, I was planning on using it sometime before myself but currently am not.

File: envs/arm_binpick.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+from brax.io import mjcf
+import mujoco
+import jax
+from jax import numpy as jnp
+
+
+# Observation space: 46 dims for the actual observation, 7 dims for goal position. See _get_obs() for details.
+# Action space:      9 dimensional, [-1, 1] each. The first 7 correspond to joint target angles (each joint has a different range, see
+#                    convert_action). The last 2 correspond to gripper finger positions.
+class ArmBinpick(PipelineEnv):
```
[the remark this answers is not in the document]
Comment (SimpleGeometry) [reply envs/arm_binpick.py:12]: Noted, will try to think about how to refactor this. Possible they can all go into one file even, or if not let me see what I can refactor into a separate file.

[the remark this answers is not in the document]
Comment (SimpleGeometry) [reply envs/arm_binpick.py:12]: I know your initial suggestion might have been to refactor shared components into a single file, but it looked reasonable to refactor the entire files together into one file. Could you let me know if you think it's better this way or if I should still refactor it?

**Review by MichalBortkiewicz — APPROVED** [review]
Looks

**Pull request merged**